adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -1191,9 +1201,153 @@ private case class GenericGenomicRDD[T](
 
   protected def replaceRdd(
     newRdd: RDD[T],
-    newPartitionMap: Option[Array[Option[(ReferenceRegion, ReferenceRegion)]]] = None): GenericGenomicRDD[T] = {
+    isSorted: Boolean = false,
+    preservesPartitioning: Boolean = false): GenericGenomicRDD[T] = {
+
+    copy(rdd = newRdd)
+  }
+}
+
+private[rdd] object SortedGenomicRDD {
+
+  def touchFile(sc: SparkContext,
+                pathName: String) {
+
+    val path = new Path(pathName)
+    val fs = path.getFileSystem(sc.hadoopConfiguration)
+
+    val os = fs.create(path)
+    os.close()
+  }
+
+  def touchSortedMetadataFile[T, U <: GenomicRDD[T, U]](gRdd: GenomicRDD[T, U],
+                                                        sc: SparkContext,
+                                                        fileName: String) {
+
+    // is this file sorted?
+    gRdd match {
+      case _: SortedGenomicRDD[T, U] => {
+        touchFile(sc, "%s/_sorted".format(fileName))
+      }
+      case _ => { /* no-op */ }
+    }
+  }
 
-    copy(rdd = newRdd, optPartitionMap = newPartitionMap)
+  def partitionersAreCompatible(leftRdd: SortedGenomicRDD[_, _],
+                                rightRdd: SortedGenomicRDD[_, _],
+                                optPartitions: Option[Int]): Boolean = {
+
+    if (leftRdd.partitioner.numPartitions == rightRdd.partitioner.numPartitions) {
+      optPartitions.fold(true)(p => leftRdd.partitioner.numPartitions == p) &&
+        ((leftRdd.partitioner, rightRdd.partitioner) match {
+          case (IndexedGenomicRangePartitioner(leftBounds, _),
+            IndexedGenomicRangePartitioner(rightBounds, _)) => {
+            leftBounds.zip(rightBounds).forall(p => p._1 == p._2)
+          }
+          case (LexicographicalGenomicRangePartitioner(leftBounds),
+            LexicographicalGenomicRangePartitioner(rightBounds)) => {
+            leftBounds.zip(rightBounds).forall(p => {
+              p._1.referenceName == p._2.referenceName &&
+                p._1.start == p._2.start
```

Comment:
This won't solve the problem of determining that the data is copartitioned properly, correct? We are doing overlap joins, so the starts being equivalent doesn't actually tell us whether all data that should be joined together is colocated together. Please let me know if I've missed something.